cli/kubectl_kadalu/logs.py

```diff
@@ -1,44 +1,106 @@
 """
 'logs' subcommand for kubectl-kadalu CLI tool
 """
-#To prevent Py2 to interpreting print(val) as a tuple.
+# To prevent Py2 to interpreting print(val) as a tuple.
 from __future__ import print_function
 
+import sys
+import argparse
 import utils
 
+
 # TODO: provide 'hint' based on grepping error logs. That way, users get to
 #       check for few common errors easily
 
 def set_args(name, subparsers):
     """ add arguments to argparser """
-    # TODO: in future allow users to pass optional pod names
-    # TODO: make 'all-containers' optional based on container name (along with pod name).
     # TODO: provide options to pass options to kubectl logs (like tail, container name etc)
-    parser = subparsers.add_parser(name)
+
+    parser = subparsers.add_parser(
+        name,
+        formatter_class=argparse.RawTextHelpFormatter
+    )
+
+    arg = parser.add_argument
+
+    arg("-p", "--podname",
+        help="Specify pod name to get log info")
+
+    arg("-c", "--container",
+        help="Specify container name to get log info.\n"
+             "To be used along with '--podname'")
+
+    arg("-A", "--allcontainers", action="store_true",
+        help=("Show logs of all containers"
+              "of a particular pod.\n"
+              "To be used along with '--podname'"))
+
     utils.add_global_flags(parser)
 
 
-def validate(_args):
-    """No validation available"""
-    return
+def validate(args):
+    """Validate optional arguments"""
+
+    if args.container and not args.podname:
+        print("Specify pod name for the container '%s'." %args.container)
+        sys.exit(1)
+
+    if args.allcontainers and not args.podname:
+        print("Specify pod name for displaying of all containers.\n"
+              "Use without any optional arguments to get logs of "
+              "all pods and all of it's containers.")
+        sys.exit(1)
 
+    if args.allcontainers and args.container:
+        print("Cannot use both --allcontainers & --container together. "
+              "Choose any one of the above options.")
+        sys.exit(1)
+
+
+def fetch_pod_logs(args):
+    """ Get logs for a particular pod """
+    try:
+        # Set default to '--allcontainers' to avoid exception
+        container_name = ""
+        if args.allcontainers:
+            container_name = "--all-containers"
+        elif args.container:
+            container_name = args.container
```

Comment:
I think args.container is working 
```bash
vatsa@vatsa-VirtualBox:~/kadalu/cli/build$ ./kubectl-kadalu logs -p kadalu-csi-provisioner-0 -c csi-resizer
----- (Kadalu Namespace) kadalu-csi-provisioner-0 -----
I0312 10:25:43.083947       1 main.go:79] Version : v1.0.0-0-gb22717d-dirty
I0312 10:25:43.349066       1 connection.go:153] Connecting to unix:///var/lib/csi/sockets/pluginproxy/csi.sock
I0312 10:25:43.420456       1 common.go:111] Probing CSI driver for readiness
W0312 10:25:43.423725       1 metrics.go:142] metrics endpoint will not be started because `metrics-address` was not specified.
I0312 10:25:43.528827       1 controller.go:117] Register Pod informer for resizer kadalu
I0312 10:25:43.553227       1 controller.go:241] Starting external resizer kadalu

```